multinet/db.py

```diff
@@ -58,42 +58,41 @@
     port=int(os.environ.get("ARANGO_PORT", "8529")),
     protocol=os.environ.get("ARANGO_PROTOCOL", "http"),
 )
-restricted_keys = {"_rev", "_id"}
 
 
-def db(name: str) -> StandardDatabase:
+def db(name: str, readonly: bool = True, immediate: bool = False) -> StandardDatabase:
     """Return a handle for Arango database `name`."""
-    return arango.db(
-        name,
-        username="root",
-        password=os.environ.get("ARANGO_PASSWORD", "letmein"),
-        verify=True,
+
+    username = "readonly" if readonly else "root"
+    password = (
+        os.environ.get("ARANGO_READONLY_PASSWORD", "letmein")
+        if readonly
+        else os.environ.get("ARANGO_PASSWORD", "letmein")
     )
 
+    return arango.db(name, username=username, password=password, verify=immediate)
 
-def read_only_db(name: str) -> StandardDatabase:
-    """Return a read-only handle for the Arango database `name`."""
-    return arango.db(
-        name,
-        username="readonly",
-        password=os.environ.get("ARANGO_READONLY_PASSWORD", "letmein"),
-        verify=True,
-    )
+
+@lru_cache()
+def system_db(readonly: bool = True) -> StandardDatabase:
+    """Return the singleton `_system` db handle."""
+    return db("_system", readonly)
 
 
+# TODO: Fix/remove this
```

Comment:
What needs to be checked/removed? If appropriate please file an issue and link here, or remove the comment.